internal/client/api/leaves_test.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package api
+
+import (
+	"testing"
+
+	"github.com/ChainSafe/gossamer/internal/primitives/core/hash"
+	"github.com/ChainSafe/gossamer/internal/primitives/database"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLeafSet_ImportWorks(t *testing.T) {
+	set := NewLeafSet[uint32, uint32]()
+	set.Import(0, 0, 0)
+
+	set.Import(1_1, 1, 0)
```

Comment:
nit/I personally don't think the underscore syntax makes this easier to read, so would prefer to not use since the numbers we are dealing with are small

Reply:
the underscore syntax is to denote lineage of the hashes to the parent hash.  So a parent of `1_1` is `1`.

Reply:
ahhh i see this makes sense, thanks!